packages/json-parser-products/src/writer.js

```diff
@@ -0,0 +1,100 @@
+import archiver from 'archiver'
+import EmitOnce from 'single-emit'
+import fs from 'fs'
+import json2csv from 'json2csv'
+import path from 'path'
+import tmp from 'tmp'
+import slugify from './utils'
+
+// Accept a highland stream and write the output to a single file
+export function writeToSingleCsvFile (productStream, output, logger, headers) {
+  const headersString = headers.map(header => `"${header.trim()}"`).join(',')
+  output.write(`${headersString}\n`) // Write headers first
+  productStream
+    .each((product) => {
+      const csvData = json2csv({
+        data: product,
+        fields: headers,
+        hasCSVColumnTitle: false,
+      })
+      output.write(`${csvData}\n`)
+    })
+    .done(() => {
+      if (output !== process.stdout) output.end()
+
+      logger.info('All products have been written to CSV file')
+    })
+}
+
+// Accept a highland stream and write the output to multiple files per
+// product type, then compress all files to a zip file
+export function writeToZipFile (productStream, output, logger) {
+  const tmpDir = tmp.dirSync({ unsafeCleanup: true }).name
+  let currentProductType
+  let fileName
+  let filePath
+  let fileStream
```

Comment:
No need to make `fileStream`, `filePath`, `fileName` global or defined with `let`

Reply:
`fileStream` is needed with the let